### PR TITLE
Refact openstack security rule plugin tests

### PR DESCRIPTION
**Pull request by ANKerD**
This change refactors Openstack's security rule plugin as well as its unit tests. To test this PR: check out code; configure an Openstack cloud in your RAS environment; test security rule operations via a client; run security rule unit tests.

### CODE REVIEW

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java [hunk 1]
```diff
@@ -1,258 +1,512 @@
...
-        Mockito.verify(this.openStackHttpClient, Mockito.times(1)).
-                doGetRequest(Mockito.anyString(), Mockito.any(OpenStackV3User.class));
+        // verify
+        Mockito.verify(spyPublicIpOrder, Mockito.times((testUtils.RUN_ONCE))).getType();
+        Mockito.verify(spyPublicIpOrder, Mockito.times((testUtils.RUN_ONCE))).getInstanceId();
+        Assert.assertEquals(expectedGroupName, actualGroupName);
     }
 
-    //test case: Tests get security rule from json response
+    // test case: given a non-supported Order it should throw InvalidParameterException
```
Comment (pauloewerton) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: nit: unsupported*

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java [hunk 2]
```diff
@@ -1,258 +1,512 @@
...
-    public void validate() {
-        Mockito.validateMockitoUsage();
+        this.clientMock = Mockito.mock(OpenStackHttpClient.class);
+        this.plugin.setClient(this.clientMock);
+        this.cloudUser = new OpenStackV3User(FAKE_USER_ID, FAKE_NAME, FAKE_TOKEN_VALUE, FAKE_PROJECT_ID);
+
+        this.majorOrder = testUtils.createNetworkOrder(testUtils.getLocalMemberId(), testUtils.FAKE_REMOTE_MEMBER_ID);
     }
 
     // test case: The http client must make only 1 request
```
Comment (pauloewerton) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: nit: maybe we could improve this comment a little?

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java [hunk 3]
```diff
@@ -1,258 +1,512 @@
...
+
+        Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE))
+                .doPostRequest(Mockito.anyString(), Mockito.anyString(), Mockito.eq(cloudUser));
+
+        PowerMockito.verifyStatic(CreateSecurityRuleResponse.class);
+        CreateSecurityRuleResponse.fromJson(Mockito.eq(ANY_STRING));
     }
 
-    // test case: Tests if an exception will be thrown in case that openstack raise an error in security rule request.
+    // test case: Check if the method makes the expected calls
```
Comment (pauloewerton) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: ditto

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java [hunk 4]
```diff
@@ -1,258 +1,512 @@
...
+        Mockito.verify(spyPublicIpOrder, Mockito.times((testUtils.RUN_ONCE))).getType();
+        Mockito.verify(spyPublicIpOrder, Mockito.times((testUtils.RUN_ONCE))).getInstanceId();
+        Assert.assertEquals(expectedGroupName, actualGroupName);
     }
 
-    //test case: Tests get security rule from json response
+    // test case: given a non-supported Order it should throw InvalidParameterException
+    @Test(expected = InvalidParameterException.class)
+    public void testRetrieveSecurityGroupNameWithInvalidOrder()
+            throws InvalidParameterException {
```
Comment (pauloewerton) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: nit: unnecessary line break

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java [hunk 5]
```diff
@@ -1,258 +1,512 @@
...
 
     @Before
     public void setUp() throws InvalidParameterException, UnexpectedException {
-        PropertiesHolder propertiesHolder = PropertiesHolder.getInstance();
-        this.properties = propertiesHolder.getProperties();
-        this.properties.put(NETWORK_NEUTRONV2_URL_KEY, DEFAULT_NETWORK_URL);
-
+        this.testUtils.mockReadOrdersFromDataBase();
         String confFilePath = HomeDir.getPath() + SystemConstants.CLOUDS_CONFIGURATION_DIRECTORY_NAME + File.separator
                 + "default" + File.separator + SystemConstants.CLOUD_SPECIFICITY_CONF_FILE_NAME;
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: Use the constant DEFAULT_CLOUD_NAME  from TestUtils class.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java [hunk 6]
```diff
@@ -1,258 +1,512 @@
...
+                Mockito.mock(GetSecurityGroupsResponse.SecurityGroup.class);
+        Mockito.when(securityGroupMock.getId()).thenReturn(id);
+        return securityGroupMock;
+    }
 
-        NetworkInstance networtkInstanceExcepted = new NetworkInstance(order.getId());
-        order.setInstanceId(networtkInstanceExcepted.getId());
-        return order;
+    private SecurityRule createEmptySecurityRule() {
+        return new SecurityRule(SecurityRule.Direction.OUT, 0, 0, "0.0.0.0/0 ", SecurityRule.EtherType.IPv4, SecurityRule.Protocol.TCP);
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: Use constants to define the CIDR and ports from the rule.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulePlugin.java [hunk 7]
```diff
@@ -49,39 +48,31 @@ public String requestSecurityRule(SecurityRule securityRule, Order majorOrder,
                                       OpenStackV3User cloudUser) throws FogbowException {
             CreateSecurityRuleResponse createSecurityRuleResponse = null;
```
Comment (wassoares) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulePlugin.java]: Correct the indentation of this line.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java [hunk 8]
```diff
@@ -1,258 +1,512 @@
...
+
+    private OpenStackSecurityRulePlugin plugin;
+    private OpenStackV3User cloudUser;
     private Properties properties;
-    private OpenStackHttpClient openStackHttpClient;
+    private OpenStackHttpClient clientMock;
+    private NetworkOrder majorOrder;
 
     @Before
     public void setUp() throws InvalidParameterException, UnexpectedException {
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: Remove unused exception throws.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java (hunk 1 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: fixed.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java (hunk 2 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: Okay

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java (hunk 3 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: Ok.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java (hunk 4 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: Indeed.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java (hunk 5 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: done.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java (hunk 6 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: Changed.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java (hunk 8 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java]: Removed.

**Pull request merged**